File: Merge_Sorted_Arrays/Merge_Sorted_Arrays.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    std::vector<int> v1 = {1, 3, 5};
    std::vector<int> v2 = {2, 4, 6};
    std::vector<int> result(v1.size() + v2.size());
    std::merge(v1.begin(), v1.end(), v2.begin(), v2.end(), result.begin());

    for (int n : result) std::cout << n << " ";
    return 0;
}

File: Merge_Sorted_Arrays/Merge_Sorted_Arrays.java
import java.util.Arrays;

public class Merge_Sorted_Arrays {
    public static int[] merge(int[] arr1, int[] arr2) {
        int[] result = new int[arr1.length + arr2.length];
        int i = 0, j = 0, k = 0;
        while (i < arr1.length && j < arr2.length) {
            result[k++] = (arr1[i] < arr2[j]) ? arr1[i++] : arr2[j++];
        }
        while (i < arr1.length) result[k++] = arr1[i++];
        while (j < arr2.length) result[k++] = arr2[j++];
        return result;
    }
    public static void main(String[] args) {
        int[] a = {1, 3, 5}, b = {2, 4, 6};
        System.out.println(Arrays.toString(merge(a, b)));
    }
}